### PR TITLE
added api sauce + redux thunk + getBook list and rentals from api

### PR DESCRIPTION
**Pull request by lfmontesanto**
**Summary**

- Added ApiSauce + Redux Thunk

- getBookList method is now fetching data from apiary service

- getRentedBooks is also fetching data from API

- rentBook service method (missing action)

### CODE REVIEW

File: Wbooks/src/redux/store.js [hunk 1]
```diff
@@ -8,9 +9,11 @@ const reducers = {
   bookReducer
 };
 
+// Reactotron.createEnhancer(true)
+
```
Comment (lfmontesanto) [inline Wbooks/src/redux/store.js]: how can I add thunk and this at the same time?

File: Wbooks/.env [hunk 2]
```diff
@@ -0,0 +1 @@
+API_BASE_URL=http://private-anon-72e517d985-woloxrailstraining.apiary-mock.com/api/v1
```
Comment (rodrigoWolox) [inline Wbooks/.env]: Remove this file from repo and add it to .gitignore

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 node_modules
 .vscode
+Wbooks/wbooks.dev.env
```
Comment (rodrigoWolox) [inline .gitignore]: Edit this route/name to just `.env`

File: Wbooks/src/redux/book/actions.js [hunk 4]
```diff
@@ -15,12 +21,37 @@ const BookActions = {
...
-  getBookList: () => ({
-    type: actionTypes.GET_BOOK_LIST
-  }),
-  getRentedBooks: () => ({
-    type: actionTypes.GET_RENTED_BOOKS
-  }),
+  getBookList: () => async dispatch => {
+    dispatch({ type: actionTypes.GET_BOOK_LIST });
+    const response = await BookService.getBookList();
+    console.log(response);
```
Comment (rodrigoWolox) [inline Wbooks/src/redux/book/actions.js]: Delete this console

File: Wbooks/src/redux/book/actions.js [hunk 5]
```diff
@@ -15,12 +21,37 @@ const BookActions = {
...
     }
   }),
-  getBookList: () => ({
-    type: actionTypes.GET_BOOK_LIST
-  }),
-  getRentedBooks: () => ({
-    type: actionTypes.GET_RENTED_BOOKS
-  }),
+  getBookList: () => async dispatch => {
+    dispatch({ type: actionTypes.GET_BOOK_LIST });
```
Comment (rodrigoWolox) [inline Wbooks/src/redux/book/actions.js]: I think this action is doing nothing on the reducer.

File: Wbooks/src/redux/book/actions.js [hunk 6]
```diff
@@ -15,12 +21,37 @@ const BookActions = {
...
+        type: actionTypes.GET_BOOK_LIST_ERROR
+      });
+    }
+  },
+  getRentedBooks: () => async dispatch => {
+    dispatch({ type: actionTypes.GET_BOOK_LIST });
+    const response = await BookService.getRentedBooks();
+    if (response.ok) {
+      const rentedBooks = response.data.page.map(rental => rental.book);
+      console.log(rentedBooks);
```
Comment (rodrigoWolox) [inline Wbooks/src/redux/book/actions.js]: Delete this console

File: Wbooks/src/redux/book/actions.js [hunk 7]
```diff
@@ -15,12 +21,37 @@ const BookActions = {
...
+        payload: response.data.page
+      });
+    } else {
+      dispatch({
+        type: actionTypes.GET_BOOK_LIST_ERROR
+      });
+    }
+  },
+  getRentedBooks: () => async dispatch => {
+    dispatch({ type: actionTypes.GET_BOOK_LIST });
```
Comment (rodrigoWolox) [inline Wbooks/src/redux/book/actions.js]: Same as before

File: Wbooks/src/redux/book/actions.js [hunk 8]
```diff
@@ -15,12 +21,37 @@ const BookActions = {
...
+    console.log(response);
+    if (response.ok) {
+      dispatch({
+        type: actionTypes.GET_BOOK_LIST_SUCCESS,
+        payload: response.data.page
+      });
+    } else {
+      dispatch({
+        type: actionTypes.GET_BOOK_LIST_ERROR
+      });
```
Comment (rodrigoWolox) [inline Wbooks/src/redux/book/actions.js]: Remember that you can get the `response.error` or wherever the error code or message could be on the response and send it to the reducer and then get the error from any connected component to response to that particular error. Again, it's just a reminder

File: Wbooks/src/redux/store.js (hunk 1 above)
Comment (rodrigoWolox) [reply Wbooks/src/redux/store.js]: Oh, ok. import `compose`from redux and try this
```
const enhancers = [];
enhancers.push(applyMiddleware(thunk));
if (__DEV__) enhancers.push(Reactotron.createEnhancer(true));
const store = createStore(appReducer, compose(...enhancers));
```

File: Wbooks/.env (hunk 2 above)
Comment (lfmontesanto) [reply Wbooks/.env]: I did it like this originally , I don't know why it was uploaded

**Pull request merged**